x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -508,10 +527,17 @@ private void handleIndexActionAuthorizationResult(
         final Metadata metadata,
         final ActionListener<Void> listener
     ) {
+        final IndicesAccessControl indicesAccessControl = result.getIndicesAccessControl();
         final Authentication authentication = requestInfo.getAuthentication();
         final TransportRequest request = requestInfo.getRequest();
         final String action = requestInfo.getAction();
-        securityContext.putIndicesAccessControl(result.getIndicesAccessControl());
+        securityContext.putIndicesAccessControl(indicesAccessControl);
+
+        if (result.isGranted() && AuthorizationUtils.isParentAction(action)) {
```

Comment:
I propose we move the `result.isGranted` in an "assert" (or better, ditch it completely), as it's redundant here (this method is not called in case authz fails).
It's unfortunate that the `IndicesAccessControl` object even has an `isGranted` flag.

Also, more importantly,  `AuthorizationUtils.isParentAction` is very generic looking, in a very generic utils class, but it's actually checking a very specific condition: if this action can have all it's child actions "bypass" their authorization.
My suggestion is something of the sorts: `securityContext.maybeSkipChildreanActionAuthorization`.

Reply:
Good point! I'll rewrite this along with other changes.